File: 0401-0500/0429-N-ary Tree Level Order Traversal/0429-N-ary Tree Level Order Traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val = NULL;
    vector<Node*> children;

    Node() {}

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/
class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> result;
        if (root == nullptr) {
            return result;
        }
        queue<Node*> Q;
        Q.push(root);
        while (!Q.empty()) {
            int qs = Q.size();
            vector<int> row;
            for (int i = 0; i < qs; ++i) {
                Node* node = Q.front();
                Q.pop();
                row.push_back(node->val);
                for (Node* child : node->children) {
                    Q.push(child);
                }
            }
            result.push_back(row);
        }
        return result;
    }
};

File: 0401-0500/0429-N-ary Tree Level Order Traversal/0429-N-ary Tree Level Order Traversal.py
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if root is None:
            return []
        level = [root]
        result = []
        while level:
            result.append([node.val for node in level])
            level = [child for node in level for child in node.children]
        return result
